Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -486,10 +491,21 @@ void GcodeSuite::G28() {
           #endif
 
           #if ENABLED(Z_SAFE_HOMING)
-            if (TERN1(POWER_LOSS_RECOVERY, !parser.seen_test('H'))) home_z_safely(); else homeaxis(Z_AXIS);
+            // H means hold the current X/Y position when probing, if not
+            // specified, move to the define safe X/Y position before homing Z.
+            if (!parser.seen_test('H'))
+              home_z_safely();
+            else
+              homeaxis(Z_AXIS);
```

Comment:
If the position where Z0 was measured changes, this affects the behavior of `apply_leveling`/`unapply_leveling` with `ABL_PLANAR` because it uses that position as the fulcrum for tilting the bed plane. That position is a `constexpr` called `level_fulcrum`. Homing the axis at any other position with `ABL_PLANAR` therefore invalidates the previous leveling matrix and will need to force a new `G29` to be required, using a new position assigned to a non-const `level_fulcrum`.